android/source/VideoLocker/src/org/edx/mobile/view/MyCoursesListActivity.java

```diff
@@ -323,4 +337,51 @@ public void onFail(Exception ex) {
             logger.error(ex);
         }
     };
+
+    /**
+     * Checks progress of all the videos that are being downloaded.
+     * If progress of any of the downloads is 100%, then marks the video as DOWNLOADED.
+     */
+    private void doRepairData() {
+        Thread maintenanceThread = new Thread(new Runnable() {
```

Comment:
This is one time logic only when app version is changed, which mostly will happen after app update. Though database operations are already thread safe, I am moving <code>doRepairData</code> method to <code>Storage</code> class and calling it from <code>MainApplication.onCreate()</code>.